item-impl/src/main/java/com/example/auction/item/impl/ItemServiceImpl.java

```diff
@@ -71,67 +74,73 @@ public ItemServiceImpl(PersistentEntityRegistry registry, BiddingService bidding
 
 
     @Override
-    public ServiceCall<Item, Item> createItem() {
+    public ServiceCall<ItemData, Item> createItem() {
         return authenticated(userId -> item -> {
-            if (!userId.equals(item.getCreator())) {
-                throw new Forbidden("User " + userId + " can't created an item on behalf of " + item.getCreator());
-            }
+            // ItemData doesn't contain a creatorId so the Principal _is_ the creator.
             UUID itemId = UUIDs.timeBased();
-            PItem pItem = new PItem(itemId, item.getCreator(), item.getTitle(), item.getDescription(),
-                    item.getCurrencyId(), item.getIncrement(), item.getReservePrice(), item.getAuctionDuration());
-            return entityRef(itemId).ask(new PItemCommand.CreateItem(pItem)).thenApply(done -> convertItem(pItem));
+            PItemData details = Mappers.fromApi(item);
+            // TODO: remove this PItem from here and pass the 3 values in the command?
+            PItem pItem = new PItem(itemId, userId, details);
+            return entityRef(itemId)
+                    .ask(new CreateItem(pItem))
+                    .thenApply(done -> Mappers.toApi(pItem));
         });
     }
 
     @Override
-    public ServiceCall<Item, Done> updateItem(UUID id) {
+    public ServiceCall<ItemData, Item> updateItem(UUID itemId) {
         return authenticated(userId -> item -> {
-            // todo implement
-            return null;
+            // userId may not match the item's creator Id. This is validated at Command Processing in the PersistentEntity
```

Comment:
Is that the pattern we want? I'm more accustomed to authorization being validated at a boundary layer above the persistence layer... in other words, in the service implementation. The reason being that sometimes you want to allow for updates to be triggered by something other than a user request, and if the user is being validated within the persistence layer, you then need to come up with some kind of synthetic user ID or other workaround.

That might not be necessary in our case, but what is the pattern that we want to demonstrate?

Handling it at this layer would also allow for a more semantically meaningful HTTP status such as 403.